instrumentation/jboss-logmanager-mdc-1.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jbosslogmanager/v1_1/JbossLoggingEventInstrumentation.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.jbosslogmanager.v1_1;
+
+import static io.opentelemetry.instrumentation.api.log.LoggingContextConstants.SPAN_ID;
+import static io.opentelemetry.instrumentation.api.log.LoggingContextConstants.TRACE_FLAGS;
+import static io.opentelemetry.instrumentation.api.log.LoggingContextConstants.TRACE_ID;
+import static net.bytebuddy.matcher.ElementMatchers.isMethod;
+import static net.bytebuddy.matcher.ElementMatchers.isPublic;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.instrumentation.api.field.VirtualField;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer;
+import java.util.Hashtable;
+import java.util.Map;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.jboss.logmanager.ExtLogRecord;
+import org.jboss.logmanager.MDC;
+
+public class JbossLoggingEventInstrumentation implements TypeInstrumentation {
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return named("org.jboss.logmanager.ExtLogRecord");
+  }
+
+  @Override
+  public void transform(TypeTransformer transformer) {
+    transformer.applyAdviceToMethod(
+        isMethod()
+            .and(isPublic())
+            .and(named("getMdc"))
+            .and(takesArguments(1))
+            .and(takesArgument(0, String.class)),
+        JbossLoggingEventInstrumentation.class.getName() + "$GetMdcAdvice");
+
+    transformer.applyAdviceToMethod(
+        isMethod().and(isPublic()).and(named("getMdcCopy")).and(takesArguments(0)),
```

Comment:
`ExtLogRecord` does not have a method `getMdcCopy`, there's `copyMdc()` though; and anyway, is instrumenting this method needed?

Reply:
Updated, I think it is needed as otherwise people may get a copy of MDC that is inconsistent with the actual MDC they get.  But correct me if I am wrong.

Reply:
ah, I had a similar thought below as @mateuszrzeszutek, it looks like `copyMdc` copies the thread local MDC values into the object (e.g. for calling before passing the object to another thread), but it doesn't make a copy of the object itself. and since we are storing the Context in the object, this should be accessible from the other thread also.

Reply:
that makes sense, removed.